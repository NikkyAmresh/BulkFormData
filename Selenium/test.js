require("chromedriver");
const assert = require("assert");
const chai = require("chai");
const { jestSnapshotPlugin } = require("mocha-chai-jest-snapshot");
const expect = chai.expect;
chai.use(jestSnapshotPlugin());

const { Builder, Key, By, until } = require("selenium-webdriver");
const chrome = require("selenium-webdriver/chrome");
// const service = new chrome.ServiceBuilder("./chromedriver1eebc8c");
const firefox = require("selenium-webdriver/firefox");

const screen = {
  width: 640,
  height: 480,
};

let driver;

before(async function () {
  driver = await new Builder()
    .forBrowser("chrome")
    .setChromeOptions(
      new chrome.Options()
        .addArguments("--headless")
        .addArguments("--no-sandbox")
        .addArguments("--disable-dev-shm-usage")
        .windowSize(screen)
    )
    // .setChromeService(service)
    .setFirefoxOptions(new firefox.Options().headless().windowSize(screen))
    .build();
});

const baseUrl = "https://the-internet.herokuapp.com/";

var indexFunctions = require("./index");

describe("Root Tests", async () => {
  let TheIntenetServices = indexFunctions.TheIntenetServices;

  it("should open the 'The Internet Website'", async () => {
    result = await TheIntenetServices.open(driver);
    assert.equal("The Internet", result.title, "Error getting the title");
    expect(
      await result.driver.findElement(By.xpath("/html")).getText()
    ).toMatchSnapshot();
  });

  it("it should open the 'Basic Auth' link and authenticate using credentials", async () => {
    result = await TheIntenetServices.basicAuth(driver);
    const link = await result.driver.getCurrentUrl();
    assert.equal(baseUrl + "basic_auth", link, "Error getting the URL");
    const text = await result.driver
      .findElement(By.xpath('//*[@id="content"]/div/p'))
      .getText();
    assert.equal(
      "Congratulations! You must have the proper credentials.",
      text,
      "Error authenticating"
    );
    expect(
      await result.driver.findElement(By.xpath("/html")).getText()
    ).toMatchSnapshot();
  });

  it("it should open the link 'Sortable Data Tables' and sort the first table by column 'Website'", async () => {
    result = await TheIntenetServices.clickOnSortableTableLink(driver);

    const newUrl = await result.driver.getCurrentUrl();
    const newTitle = await result.driver.getTitle();

    assert.equal(baseUrl + "tables", newUrl, "Error getting the new URL");
    assert.equal("The Internet", newTitle, "Error getting the Title");
    expect(result.tableText).toMatchSnapshot();
    expect(
      await result.driver.findElement(By.xpath("/html")).getText()
    ).toMatchSnapshot();
  });
});

after(() => driver && driver.quit());
