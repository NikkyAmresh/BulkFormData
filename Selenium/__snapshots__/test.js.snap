// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root Tests it should open the 'Basic Auth' link and authenticate using credentials 1`] = `
"Basic Auth
Congratulations! You must have the proper credentials.
Powered by Elemental Selenium"
`;

exports[`Root Tests it should open the link 'Sortable Data Tables' and sort the first table by column 'Website' 1`] = `
"Last Name First Name Email Due Web Site Action
Bach Frank fbach@yahoo.com $51.00 http://www.frank.com edit delete
Doe Jason jdoe@hotmail.com $100.00 http://www.jdoe.com edit delete
Smith John jsmith@gmail.com $50.00 http://www.jsmith.com edit delete
Conway Tim tconway@earthlink.net $50.00 http://www.timconway.com edit delete"
`;

exports[`Root Tests it should open the link 'Sortable Data Tables' and sort the first table by column 'Website' 2`] = `
"Data Tables
Often times when you see a table it contains data which is sortable -- sometimes with actions that can be taken within each row (e.g. edit, delete). And it can be challenging to automate interaction with sets of data in a table depending on how it is constructed.
Example 1
No Class or ID attributes to signify groupings of rows and columns
Last Name First Name Email Due Web Site Action
Bach Frank fbach@yahoo.com $51.00 http://www.frank.com edit delete
Doe Jason jdoe@hotmail.com $100.00 http://www.jdoe.com edit delete
Smith John jsmith@gmail.com $50.00 http://www.jsmith.com edit delete
Conway Tim tconway@earthlink.net $50.00 http://www.timconway.com edit delete
Example 2
Class and ID attributes to signify groupings of rows and columns
Last Name First Name Email Due Web Site Action
Smith John jsmith@gmail.com $50.00 http://www.jsmith.com edit delete
Bach Frank fbach@yahoo.com $51.00 http://www.frank.com edit delete
Doe Jason jdoe@hotmail.com $100.00 http://www.jdoe.com edit delete
Conway Tim tconway@earthlink.net $50.00 http://www.timconway.com edit delete
Powered by Elemental Selenium"
`;

exports[`Root Tests should open the 'The Internet Website' 1`] = `
"Welcome to the-internet
Available Examples
A/B Testing
Add/Remove Elements
Basic Auth (user and pass: admin)
Broken Images
Challenging DOM
Checkboxes
Context Menu
Digest Authentication (user and pass: admin)
Disappearing Elements
Drag and Drop
Dropdown
Dynamic Content
Dynamic Controls
Dynamic Loading
Entry Ad
Exit Intent
File Download
File Upload
Floating Menu
Forgot Password
Form Authentication
Frames
Geolocation
Horizontal Slider
Hovers
Infinite Scroll
Inputs
JQuery UI Menus
JavaScript Alerts
JavaScript onload event error
Key Presses
Large & Deep DOM
Multiple Windows
Nested Frames
Notification Messages
Redirect Link
Secure File Download
Shadow DOM
Shifting Content
Slow Resources
Sortable Data Tables
Status Codes
Typos
WYSIWYG Editor
Powered by Elemental Selenium"
`;
